### ISSUE TITLE
Add support for Google authentication and Google Drive

### ISSUE THREAD

**Issue by ericfitz** [OWASP/threat-dragon#1088]
**Describe what problem your feature request solves**:
I would like to use the Threat Dragon tool in a non-GitHub environment.  That means that I'd like to use Google Drive as a data store, and authenticate to the Threat Dragon application (and to Google Drive) using Google authentication.

**Describe the solution you'd like**:
I'd like to be able to configure an instance of Threat Dragon to use [Google Sign-In](https://developers.google.com/identity/sign-in/web/sign-in), e.g. so I can authenticate as "myaccount@gmail.com".

I'd also like to be able to configure an instance of Threat Dragon to use a Google Drive folder as a store for models, rather than a GitHub repo.

**Additional context**:
Related to [issue 426](https://github.com/OWASP/threat-dragon/issues/426)

**Comment by ericfitz** [comment]
I've posted a [bounty](https://www.upwork.com/jobs/~021839022842033686517) for this issue.

**Comment by hasan-tec** [comment]
Thanks for sharing the details @ericfitz! I'd like to express interest in working on this feature enhancement for Google authentication and Drive integration.

To clarify, I'll implement Google OAuth 2.0 for sign-in, allowing users to authenticate via their Google accounts (e.g., "myaccount@gmail.com"). I'll also set up Google Drive as a data store, enabling model files to be uploaded and managed in Drive rather than a GitHub repository.

Please feel free to provide any specific requirements or preferences regarding the integration. I'll review the codebase and propose an implementation approach. Let me know if there's any additional information you would like to share before I proceed.

Thanks!

**Comment by ericfitz** [comment]
Thanks hasan-tec - you'll have to go through Upwork to get paid - the link to my job is there!

Basically I just want:
1. A working, standard "sign in with Google" button that I can enable via configuration (and OAuth app registration with Google).  Authentication needs to be done by Google (OAuth 2.0); we should not be collecting Google creds in the app.

2. The ability to set configuration to allow/require Google Drive as a store for model files.

3. A user experience to select the Google Drive folder in which to load/store models.

**Comment by jgadsden** [comment]
Thank you @ericfitz for posting the bounty and raising this issue

**Comment by hasan-tec** [comment]
will i get paid for this? as it was in upworks job? 500$?

**Comment by ericfitz** [comment]
Hey Hasan-tec, I posted the bounty on upworks but have not hired a candidate yet.  I am interviewing now.  It's easy to pay through upworks.  Do you have a way to chat?

**Comment by hasan-tec** [comment]
Yes you can, we have an agency on Upwork. You can come over to my agency, ByteForth, I have a professional team, you can hire us and then it will be easy for you to pay through Upwork. I have top 2 top rated Upwork freelancer and myself a rising talent on Upwork as well. The payment will be discussed there, don't worry it will be around 500, we don't overcharge, so don't worry about that. I will send you the link you can directly contaxt us, and then make a contract.

**Comment by hasan-tec** [comment]
https://www.upwork.com/agencies/byteforth/

**Comment by hasan-tec** [comment]
I have applied to your Upwork job post, and my team is ready for the interview anytime you want, and are really excited and ready for this job.

**Comment by hasan-tec** [comment]
@jgadsden you didnt had to unassign me, lmao, all g,

**Comment by hasan-tec** [comment]
Hiya Eric,
did you decline my Upwork job proposal?

**Comment by bryng001** [comment]
Hi Eric, here is an update of the progress:
9/27/24
What I did yesterday: 
Added the google sign in feature using OAuth.
PR - https://github.com/OWASP/threat-dragon/pull/1092
What I will do today: 
Will analyze the entire logic of this project and prepare to add Google Drive feature.
Roadblocks: No
Estimated completion: 35%
Project delivery status: Green

I don’t have any questions so far. Do let me know you if have additional requests. Thanks!
